tests/network/link_state.go

```diff
@@ -0,0 +1,229 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2025 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	. "kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libnet"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const (
+	defaultInterfaceName = "default"
+	bridgeInterfaceName2 = "bridge2"
+	bridgeInterfaceNAD2  = "bridge2nad"
+	bridgeInterfaceName3 = "bridge3"
+	bridgeInterfaceNAD3  = "bridge3nad"
+	regexLinkStateUP     = "'state[[:space:]]+UP'"
+	regexLinkStateDOWN   = "'state[[:space:]]+DOWN'"
+	ipLinkTemplate       = "ip -one link | grep %s | grep -E %s\n"
+)
+
+var _ = SIGDescribe("interface state up/down", func() {
+
+	var (
+		vmName           types.NamespacedName
+		mac1, mac2, mac3 net.HardwareAddr
+	)
+	BeforeEach(func() {
+		var ifaceDefault *v1.Interface
+		var iface2 *v1.Interface
+
+		var err error
+		_, err = libnet.CreateNetAttachDef(context.Background(), testsuite.GetTestNamespace(nil),
```

Comment:
Please consider having a `testNamespace` variable that will cache the result of `testsuite.GetTestNamespace(nil)`.